api/kogito-api/src/main/java/org/kie/kogito/process/ProcessInstance.java

```diff
@@ -169,7 +187,22 @@
 
     /**
      * Returns the process milestones
+     *
      * @return All the process milestones with their current status
      */
     Collection<Milestone> milestones();
+
+    /**
+     * Returns the SubProcesses in the process
+     *
+     * @return All the process {@link SubProcess} with their current status
+     */
+    Collection<SubProcess> subProcesses();
+
+    /**
+     * Returns the process adHocFragments
+     *
+     * @return All the {@link AdHocFragment} in the process
+     */
+    Collection<AdHocFragment> getAdHocFragments();
```

Comment:
I think to align with the other methods naming we would call this: 
```suggestion
    Collection<AdHocFragment> adHocFragments();
```